### PR TITLE
Fixed #762: Improved shallow options documentation

### LINKED ISSUE #762: Improve `shallow` docs for options
The docs for `shallow`: https://github.com/airbnb/enzyme/blob/master/docs/api/shallow.md don't list supported fields of the `options` object. They do show that you can provide a `context` option, but *all* supported options aren't clearly illuminated. They are missing at least one: `lifecycleExperimental`

### PR DESCRIPTION
**Pull request by niveditn**
Fixes #762:

Added details of the following `options` parameters for `shallow` API:

- `disableLifecycleMethods`
- `lifecycleExperimental`

This should update the GitBook html documentation to include all available options:
http://airbnb.io/enzyme/docs/api/ShallowWrapper/shallow.html

### CODE REVIEW

**Comment by nxneha** [comment]
@ljharb I just rebased this on the latest master to ensure tests are passing. It would be great if you could please review. Thanks!

**Comment by NiveditN** [comment]
@ljharb I just rebased this on the latest master to ensure tests are passing. It would be great if you could please review. Thanks!

**Comment by NiveditN** [comment]
Hey @ljharb, I can help with the updates if you could please tell me what needs to be done.

File: docs/api/ShallowWrapper/shallow.md [hunk 1]
```diff
@@ -8,7 +8,13 @@ NOTE: can only be called on wrapper of a single node.
 #### Arguments
 
 1. `options` (`Object` [optional]):
-- `options.context`: (`Object` [optional]): Context to be passed into the component
+  - `options.context`: (`Object` [optional]): Context to be passed into the component
+  - `options.disableLifecycleMethods`: (`Boolean` [optional]): If set to true, `componentDidMount`
+is not called on the component, and `componentDidUpdate` is not called after
+[`setProps`](ShallowWrapper/setProps.md) and [`setContext`](ShallowWrapper/setContext.md). Default to `false`.
+  - `options.lifecycleExperimental`: (`Boolean` [optional]): If set to true, the entire lifecycle
```
Comment (koba04) [inline docs/api/ShallowWrapper/shallow.md]: Enzyme v3 enables `lifecycleExperimental` by default. So this can be removed.

**Comment by niveditn** [comment]
@koba04, @ljharb As suggested, I have updated the fix to remove `lifecycleExperimental` details for v3. Please review and let me know if any further changes are needed. Thanks!

**Pull request merged**